src/harness/fourslash.ts

```diff
@@ -4654,12 +4735,33 @@ namespace FourSlashInterface {
         newContent: string;
     }
 
-    export type ExpectedCompletionEntry = string | { name: string, insertText?: string, replacementSpan?: FourSlash.Range };
+    export type ExpectedCompletionEntry = string | {
+        readonly name: string,
+        readonly insertText?: string,
+        readonly replacementSpan?: FourSlash.Range,
+        readonly hasAction?: boolean, // If not specified, will assert that this is false.
+        readonly kind?: string, // If not specified, won't assert about this
+        readonly text: string;
+        readonly documentation: string;
+        readonly sourceDisplay?: string;
+    };
     export interface CompletionsAtOptions extends Partial<ts.UserPreferences> {
         triggerCharacter?: string;
         isNewIdentifierLocation?: boolean;
     }
 
+    export interface VerifyCompletionsOptions {
+        readonly at?: Many<string>;
+        readonly isNewIdentifierLocation?: boolean;
+        readonly are?: Many<ExpectedCompletionEntry>;

```

Comment:
I find the name "at" a little unclear and the name "are" very unclear.

Reply:
`at` could be `atMarker` -- not sure what to replace `are` with -- it's what the completions are. `areExactly`?

Reply:
I figured out what they meant by reading the tests, but I still find the names unclear.  I would have expected something like `location` or `marker` for `at` and something like `all` or `exact` for `are` (or a flag indicating that `includes` is exhaustive).

Reply:
`marker` and `exact` sound good.